sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -99,7 +99,8 @@ case class CatalogTablePartition(
     spec: CatalogTypes.TablePartitionSpec,
     storage: CatalogStorageFormat,
     parameters: Map[String, String] = Map.empty,
-    stats: Option[CatalogStatistics] = None) {
+    stats: Option[CatalogStatistics] = None,
+    schema: Option[StructType] = None) {
```

Comment:
The partition schema is stored in `CatalogTable `. I am not very clear what is the exception you got. 

@dongjoon-hyun Could you help @liutang123 investigate the issue?

Reply:
Sure, @gatorsmile . I'll take a look during weekend.

Reply:
Some times, partition's schema is different from the table's.